### PR TITLE
EOS-23494:  Re-deployment fails on VM after running cortx_setup cluster reset --type all

### PR DESCRIPTION
**Pull request by tanujashinde0405**

### CODE REVIEW

File: srv/components/ha/corosync-pacemaker/reset.sls [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  service.dead:
+    - name: {{ serv }}
+    - enable: False
+{% endfor %}
+
+Remove pcs package:
+  cmd.run:
+    - name: "rpm -e --nodeps pcs pacemaker corosync fence-agents-ipmilan"
+
+# Remove configuration directory:
```
Comment (nitin-seagate) [inline srv/components/ha/corosync-pacemaker/reset.sls:35]: Please confirm if we have to remove config directory for services
`"corosync", "pacemaker", "pcsd"`

File: srv/components/misc_pkgs/consul/reset.sls [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+include:
+  - components.misc_pkgs.consul.stop
+
+# Remove Consul:
```
Comment (nitin-seagate) [inline srv/components/misc_pkgs/consul/reset.sls:21]: Why are we not removing only consul without Hare dependency?

File: srv/components/system/teardown.sls [hunk 3]
```diff
@@ -20,10 +20,6 @@ Remove cortx-py-utils dependencies:
     - requirements: salt://components/system/files/cortx_py_utils_requirements.txt
     - bin_env: /usr/bin/pip3
 
-Remove cortx-py-utils pip package:
```
Comment (nitin-seagate) [inline srv/components/system/teardown.sls:23]: Why are we removing this from teardown? I think we are uninstalling all the RPMs for teardown right...

**Review by nitin-seagate — CHANGES_REQUESTED** [review]

File: srv/components/system/teardown.sls (hunk 3 above)
Comment (tanujashinde0405) [reply srv/components/system/teardown.sls:23]: we should not be removing foundation component rpm right?

File: srv/components/ha/corosync-pacemaker/reset.sls (hunk 1 above)
Comment (tanujashinde0405) [reply srv/components/ha/corosync-pacemaker/reset.sls:35]: We are doing it as part of the same file.

File: srv/components/system/teardown.sls (hunk 3 above)
Comment (tanujashinde0405) [reply srv/components/system/teardown.sls:23]: this should be done for pre-factory and it is already there in foundation mini provisioner call, this was redundant state which was written. so I have removed it.

**Review by 83bhp — APPROVED** [review]

**Review by nitin-seagate — APPROVED** [review]
LGTM

**Review by SomwanshiAnjali — APPROVED** [review]

**Pull request merged**